File: backend/coffee-inventory-api/Data/BrewDTO.cs
namespace API.Data;

public class BrewDTO
{
    public string Id { get; set; } = string.Empty;
    public List<UsageDTO> Usage { get; set; } = new List<UsageDTO>();
    public decimal CoffeeOut { get; set; }
    public int TimeInSeconds { get; set; }
    public string Comment { get; set; } = string.Empty;
}

public class UsageDTO
{
    public string ItemId { get; set; } = string.Empty;
    public decimal Amount { get; set; } = 0;
}

File: frontend-svelte/src/lib/services/entities/brewDTO.ts
import type {InventoryItemDTO} from "$lib/services/entities/inventoryItemDTO";

export interface BrewDTO {
    id: string,
    time: Date
    usage: UsageDTO[],
    coffeeOut: number,
    timeInSeconds: number,
    comment: string
}

export interface UsageDTO {
    itemId: string,
    amount: number
}

export interface BrewWithItems extends BrewDTO {
    usageItems: { item: InventoryItemDTO, amount: number }[],
}